lisa/operating_system.py

```diff
@@ -261,8 +502,69 @@ def _initialize_package_installation(self) -> None:
                 timeout=3600,
             )
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
-        self._node.execute(f"yum install -y {' '.join(packages)}", sudo=True)
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        command = f"yum install -y {'|'.join(packages)}"
+        if not signed:
+            command.__add__(" --nogpgcheck")
+
+        install_result = self._node.execute(command, sudo=True)
+        # yum returns exit_code=1 if package is already installed.
+        # We do not want to fail if exit_code=1.
+        if install_result.exit_code == 1:
+            self._log.info(
+                f"{' '.join(packages)} is/are already installed."
+                f" stdout: {install_result.stdout}"
+            )
+        elif install_result.exit_code != 0:
+            raise LisaException(
+                f"Failed to install {packages}." f" stdout: {install_result.stdout}"
+            )
+        else:
+            self._log.info(
+                f"{packages} is/are installed successfully."
+                f" stdout: {install_result.stdout}"
+            )
+
+    def _package_exists(self, package: str, signed: bool = True) -> bool:
+        command = f"yum list installed | grep -Ei {package}"
+        result = self._node.execute(command, sudo=True)
+        if result.exit_code == 0:
+            return True
+
+        return False
+
+    def _get_os_version(self) -> OsVersion:
+        if self._os_version:
```

Comment:
The fedora has an implementation, why it needs to implement Redhat too? If the fedora-release works, we don't need Redhat version. There are a lot of derives of Linux, we don't support all. We just support which is enough for our testing, what we meet.

Reply:
Fedora implementation is checking info  from /etc/fedora-release file.
Redhat implementation is checking info from  /etc/redhat-release file.
We cannot always be sure that we will get /etc/fedora-release file on Redhat distro.